lib/charms/opensearch/v0/helper_conf_setter.py

```diff
@@ -57,9 +60,73 @@ class YamlConfigSetter:
 
     def __init__(self, base_path: str = None):
         """base_path: if set, where to look for files relatively on "load/put/delete" methods."""
-        self.yaml = YAML()
         self.base_path = self.__clean_base_path(base_path)
 
+    @abstractmethod
+    def load(self, config_file: str) -> Dict[str, any]:
+        """Load the content of a YAML file."""
+        pass
+
+    @abstractmethod
+    def put(
+        self,
+        config_file: str,
+        key_path: str,
+        val: any,
+        sep="/",
+        output_type: OutputType = OutputType.file,
+        inline_array: bool = False,
+        output_file: str = None,
+    ) -> Dict[str, any]:
+        """Add or update the value of a key (or content of array at index / key) if it exists."""
+        pass
```

Comment:
This was raised in a previous PR. So, the way I designed this class is to have self contained operations - in the context of OpenSearch YAML config files, the goal being to provide implementations of what "put" means in the context of "VM" and "K8s" charms. To ultimately write on the file system or a storage in a single call.

This helper is not to manipulate and wrangle YAML objects for the sake of having a transformed object - as we do not need those in the charms' logic.
 
Using a Builder pattern here would require maintaining a state object, which I am trying to avoid as it would violate the self-containment of operations I'm establishing + this helper class is also expected to be used from shell scripts in various places such as snaps, replacing the `yq` utility sometime soon.